test/mlir/conversion/onnx_to_tosa/NN/MaxPoolSingleOut.mlir

```diff
@@ -0,0 +1,103 @@
+// RUN: onnx-mlir-opt --shape-inference --convert-onnx-to-tosa %s -split-input-file | FileCheck %s
+
+/// Test the default behavior of Max Pool with no padding
+func.func @test_default_maxpoolsingleout(%arg0 : tensor<5x5x32x32xf32>) -> tensor<5x5x30x30xf32> {
+  %0 = "onnx.MaxPoolSingleOut"(%arg0) {kernel_shape = [3,3]} : (tensor<5x5x32x32xf32>) -> tensor<5x5x30x30xf32>
```

Comment:
Did you run the pass with the `--debug` flag and checked if ONNX or TOSA emitted shape warnings? Just out of curiosity how you got all the shapes. That was the main reason I adapted your dynamic shape + infer combination, but you set all shapes explicitly

Reply:
If you are talking about the ONNX mlir code, I got it from other tests mostly. 
Though, I ran my tests with `--debug` and saw some warning. I'm investigating them right now, but I also saw that fully relying on shape inference failed in TOSA, hence why I'm calling `rewriter.create` to create the MaxPool and not `CreateOpAndInfer`

Reply:
Update on that, it was a poorly written test that was generating warning. So, no warning, now that it's fixed